Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedCollectionOperations.swift

```diff
@@ -1,7 +1,81 @@
 import Foundation
+@testable import struct MongoSwift.AggregateOptions
 @testable import struct MongoSwift.FindOptions
 import MongoSwiftSync
 
+struct UnifiedAggregate: UnifiedOperationProtocol {
+    /// Aggregation pipeline.
+    let pipeline: [BSONDocument]
+
+    /// Options to use for the operation.
+    let options: AggregateOptions
+
+    /// Optional identifier for a session entity to use.
+    let session: String?
+
+    private enum CodingKeys: String, CodingKey, CaseIterable {
+        case pipeline, session
+    }
+
+    static var knownArguments: Set<String> {
+        Set(
+            CodingKeys.allCases.map { $0.rawValue } +
+                AggregateOptions.CodingKeys.allCases.map { $0.rawValue }
+        )
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.pipeline = try container.decode([BSONDocument].self, forKey: .pipeline)
+        self.session = try container.decodeIfPresent(String.self, forKey: .session)
+        self.options = try decoder.singleValueContainer().decode(AggregateOptions.self)
+    }
+}
+
+struct UnifiedCreateIndex: UnifiedOperationProtocol {
+    /// The name of the index to create.
+    let name: String
+
+    /// Keys for the index.
+    let keys: BSONDocument
+
+    /// Optional identifier for a session entity to use.
+    let session: String?
+
+    static var knownArguments: Set<String> {
+        ["name", "keys", "session"]
+    }
+}
+
+struct UnifiedBulkWrite: UnifiedOperationProtocol {
+    /// Writes to perform.
+    let requests: [WriteModel<BSONDocument>]
```

Comment:
Instead of relying on `WriteModel`'s `Decodable` implementation for both the old runner and the unified one, what do you think about defining a new intermediate type that has a a unified-runner-specific `Decodable` implementation and using that here instead? Then we don't need to mix implementations.

Reply:
it does seem a bit cleaner. done